### PR TITLE
[FEAT]: Implement an array sorting script

### LINKED ISSUE #105: [FEAT]: Implement an array sorting script
## 📘 Description

We need to implement one sorting algorithms to provide flexible and efficient array sorting capabilities. This implementation will include one of these: Bubble Sort, Quick Sort, and Merge Sort algorithms, allowing users to choose the most appropriate method based on their specific use case.

## ✅ Acceptance Criteria

Implement one of these sorting algorithms:

- Bubble Sort: Simple but inefficient, good for educational purposes
- Quick Sort: Efficient general-purpose sorting
- Merge Sort: Stable sorting with guaranteed performance

The selected algorithm should:

- Accept an array of numbers as input
- Support both ascending and descending order
- Return the sorted array
- Handle edge cases (empty arrays, single elements)

## 🌎 References

- [Alexandria Sorting Package](https://github.com/keep-starknet-strange/alexandria/tree/main/packages/sorting)

## 📜 Additional Notes

- ⚠ **Do not apply until the ODBoost begins.**
- ⚠ **Read our contribution guidelines before applying.**

### PR DESCRIPTION
**Pull request by martinvibes**
# 📝 Pull Request Title
## [FEAT]: Implement Sorting Algorithms
🛠️ Issue

Closes #105

## 📖 Description
This PR implements sorting algorithms to provide efficient array sorting capabilities. The implementation includes:

Created interface.cairo defining Sortable and SortableVec traits for consistent sorting interfaces
Implemented QuickSort algorithm with optimized pivot selection
Added Alexandria data structures as a dependency for vector operations
Structured the code in a modular way with separate files for interface and implementation
Fixed compilation issues and imports
Added proper error handling for edge cases (empty arrays, single elements)

## 🖼️ Screenshots
N/A - Implementation is code-only
📝 Additional Notes

The implementation supports both array and vector-based sorting
The code is prepared for future additions of other sorting algorithms
Tested with various input sizes and types
All files include proper documentation and type annotations

### CODE REVIEW

**Comment by martinvibes** [comment]
hey @coxmars kindly review :)

**Comment by coxmars** [comment]
Hi @martinvibes, can you add a .tool-versions file with the Scarb and Starknet foundry versions you are using?

**Comment by martinvibes** [comment]
@coxmars i have added it sir

**Review by coxmars — CHANGES_REQUESTED** [review]
Add small tests for each sorting (2 edge cases at least) to validate that they all work correctly and add a screenshot.

**Comment by martinvibes** [comment]
Alright sir know it

**Comment by martinvibes** [comment]
hello @coxmars i have written tests for them and all tests are passing :)
image

**Review by coxmars — APPROVED** [review]
Local verification done, all good. Ready for second review. @danielcdz 🎯

File: cairo/scripts/sorting/.tool-versions [hunk 1]
```diff
@@ -0,0 +1,2 @@
+scarb 2.9.2
+snforge 0.31.0
```
Comment (danielcdz) [inline cairo/scripts/sorting/.tool-versions:2]: can you add an extra line at the end of this file?

**Comment by martinvibes** [comment]
hello @danielcdz i have implemented required change :+1: 
Everything looks Good now :)

**Review by danielcdz — APPROVED** [review]
LGTM
